From f93c98b0ce4185c640ac60829b06a609913bba3d Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Fri, 18 Oct 2024 00:04:11 +0200
Subject: [PATCH] backport nv_get_kern_phys_address() changes from 555.42.02

---
 nvidia/nv-vtophys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nvidia/nv-vtophys.c b/nvidia/nv-vtophys.c
index df2a01e..fcae701 100644
--- a/nvidia/nv-vtophys.c
+++ b/nvidia/nv-vtophys.c
@@ -29,7 +29,7 @@
 NvU64 NV_API_CALL nv_get_kern_phys_address(NvU64 address)
 {
     /* direct-mapped kernel address */
-    if (virt_addr_valid(address))
+    if (virt_addr_valid((void *)address))
         return __pa(address);
 
     nv_printf(NV_DBG_ERRORS,
-- 
2.39.5

