From a6ffc23f02af1793b9ac5059fe871cdb7ee6a0ae Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Fri, 21 Feb 2025 10:21:42 +0100
Subject: [PATCH] backport module_import_ns_takes_string_literal changes from
 550.144.03

---
 conftest.sh          | 17 +++++++++++++++++
 nvidia/nvidia.Kbuild |  1 +
 2 files changed, 18 insertions(+)

diff --git a/conftest.sh b/conftest.sh
index 49003b46..ac68c360 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -5478,6 +5478,23 @@ compile_test() {
             else
                 echo "#undef NV_OF_PROPERTY_READ_VARIABLE_U32_ARRAY_PRESENT" | append_conftest "functions"
             fi
+            ;;
+
+        module_import_ns_takes_string_literal)
+            #
+            # Determine if the MODULE_IMPORT_NS macro takes a string literal
+            # or constant.
+            #
+            # Commit cdd30ebb1b9f ("module: Convert symbol namespace to
+            # string literal") changed MODULE_IMPORT_NS to take a string
+            # literal in Linux kernel v6.13.
+            #
+            CODE="
+            #include <linux/module.h>
+
+            MODULE_IMPORT_NS(DMA_BUF);"
+
+            compile_check_conftest "$CODE" "NV_MODULE_IMPORT_NS_TAKES_STRING_LITERAL" "" "functions"
         ;;
 
         devm_of_platform_populate)
diff --git a/nvidia/nvidia.Kbuild b/nvidia/nvidia.Kbuild
index 4099ae81..5f2b6aa9 100644
--- a/nvidia/nvidia.Kbuild
+++ b/nvidia/nvidia.Kbuild
@@ -193,6 +193,7 @@ NV_CONFTEST_FUNCTION_COMPILE_TESTS += get_task_ioprio
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += mdev_set_iommu_device
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += offline_and_remove_memory
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += crypto_tfm_ctx_aligned
+NV_CONFTEST_FUNCTION_COMPILE_TESTS += module_import_ns_takes_string_literal
 
 NV_CONFTEST_SYMBOL_COMPILE_TESTS += is_export_symbol_gpl_of_node_to_nid
 NV_CONFTEST_SYMBOL_COMPILE_TESTS += is_export_symbol_gpl_sme_active
-- 
2.39.5

